db/migrate/20210618104853_add_code_index_to_experiments.rb

```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class AddCodeIndexToExperiments < ActiveRecord::Migration[6.1]
+  def change
+    add_index :experiments, "('EX' || id)", name: 'index_experiments_on_experiment_code', unique: true
```

Comment:
ID is a primary key, why do you need unique constrain here?

Reply:
if you know the value is unique, it's supposed to improve performance of the index, as it can be better optimized

Reply:
`SET enable_seqscan = OFF;
EXPLAIN ANALYZE SELECT * FROM experiments WHERE ('EX' || id) ILIKE 'EX12%';
                                                         QUERY PLAN                                                         
----------------------------------------------------------------------------------------------------------------------------
 Seq Scan on experiments  (cost=10000000000.00..10000000096.14 rows=9 width=503) (actual time=0.054..0.784 rows=11 loops=1)
   Filter: (('EX'::text || (id)::text) ~~* 'EX12%'::text)
   Rows Removed by Filter: 1096
 Planning Time: 0.645 ms
 Execution Time: 0.822 ms
(5 rows)`